tracks/javascript/exercises/triangle/mentoring.md

```diff
@@ -0,0 +1,84 @@
+This is the one of the four side exercises unlocked after finishing Resistor Color. The level of understanding is vary from students. Some work on this exercise right after Resistor Color is done, others submit this exercise after solving multiple core exercise.
+
+so you need to teach at least:
+
+- [Logical Operators][ref-logical]
+
+but you might also want to teach according to students' understanding but they'll be taught in future core exercise so you don't need to stick to them.
+
+- [Rest parameters][ref-rest-parameters]
+- [Destructuring assignment][refs-destructuring-assignment]
+- [Array#every][ref-array-every]
+- [Array#some][ref-array-some]
+
+
+Also, this exercise might be the first time for some students to use class, so you gently teach them about JavaScript class if needed.
+
+### Reasonable Solutions
+
+```javascript
+export class Triangle {
+  constructor(...sides) {
+    this.sides = sides;
+  }
+
+  isEquilateral() {
+    const [a, b, c] = this.sides;
+    return this.isValid() && a == b && b == c && c == a;
+  }
+
+  isIsosceles() {
+    const [a, b, c] = this.sides;
+    return this.isValid() && (a == b || b == c || c == a);
+  }
+
+  isScalene() {
+    const [a, b, c] = this.sides;
+    return this.isValid() && a != b && b != c && c != a;
+  }
+
+  isValid(){
+    const [a, b, c] = this.sides;
+    const sidesArePositive = a > 0 && b > 0 && c > 0;
+    const validatesTriangleInequality =
+      a + b >=c && b + c >= a && c + a >= b;
+    return sidesArePositive && validatesTriangleInequality;
+  }
+}
+```
+
+A wise student might use `.sort` to simplify the Triangle Inequality rule logic.
+
+Also, some students add test cases for degenerate triangles and implement a method to check that.
+
+Students use of higher-level concepts (like number of matching sides) is also a reasonable solution and a worthy discussion point for this exercise.
+
+```javascript
+isEquilateral() {
+  return this.isValid() && this.matchingSides() === 3
+}
+
+matchingSides() { 
+  // function to calculate the number of matching sides 
+}
+```
+
+#### Approvability
+
+Some students would use `Set` to solve Triangle exercise and they're also okay if using properly.
```

Comment:
It would be nice to add the following: `We intentionally try to keep our tests free of implementation details and allow students to choose.`